src/v2/models/application.js

```diff
@@ -857,13 +857,28 @@ export default class ApplicationModel extends GenericModel {
       }));
   }
 
+  async getLocalArgoRoute(args) {
+    const routes = await this.kubeConnector.getResources((ns) => `/apis/${args.apiVersion}/namespaces/${ns}/routes`, { namespaces: [args.namespace], kind: args.kind }).catch((err) => {
+      logger.error(err);
+      throw err;
+    });
+
+    return routes.filter((route) => _.get(route, 'metadata.labels["app.kubernetes.io/part-of"]') === 'argocd' && _.get(route, 'spec.host'))[0];
```

Comment:
I don't see any check for corner cases Richie
- what happens if the routes is undefined ?
- route doesn't have that label
- you don't have to check for spec.host in here, just return the route
- what if there is not route with those props ? you get 0 elements

Reply:
Fixes added in new PR here: https://github.com/open-cluster-management/console-api/pull/318